### PR TITLE
chore: Add prettier settings and run prettier on the repo

### PR DESCRIPTION
**Pull request by AbhiPrasad**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring

## :scroll: Description

Run [prettier](https://prettier.io/) on the repo. [Based the config off the JS monorepo](https://github.com/getsentry/sentry-javascript/blob/develop/.prettierrc.json).

I did this because I attempted to remove usage of `@sentry/tracing` from the React Native SDK, since we no longer require it to be used. Kept running into formatting inconsistencies, so opened this PR.

## :bulb: Motivation and Context

Helps with consistency in formatting. We also now run this in CI. 

## :green_heart: How did you test it?

N/A

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

#skip-changelog

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,12 @@
     "build:tools": "tsc -p tsconfig.build.tools.json",
     "clean": "rimraf dist coverage",
     "test": "jest",
-    "lint": "eslint --config .eslintrc.js .",
+    "fix": "yarn fix:eslint && yarn fix:prettier",
+    "fix:eslint": "eslint --config .eslintrc.js --fix .",
+    "fix:prettier": "prettier --write \"{src,test,scripts}/**/**.ts\"",
+    "lint": "yarn lint:eslint && lint:prettier",
```
Comment (krystofwoldrich) [inline package.json:23]: Just a missing `yarn`.
```suggestion
    "lint": "yarn lint:eslint && yarn lint:prettier",
```

**Comment by krystofwoldrich** [comment]
Could you add `.prettierignore` with 

```
/test/react-native/versions
```

This path contains downloaded `create-react-native` apps with different RN versions. Prettier will fail during local runs.

**Comment by AbhiPrasad** [comment]
Done [bd54fcd](https://github.com/getsentry/sentry-react-native/pull/2950/commits/bd54fcde2ac09ade2bc7c52d9d7adc31c45ee526)

**Review by krystofwoldrich — APPROVED** [review]
@AbhiPrasad Thank you. 🙏 It's awesome to have Prettier in RN too. 🚀

**Pull request merged**